rust-lang/cargo
# Expose checksum in `cargo metadata` command

### What does this PR try to resolve?

This change exposes the `checksum` field in packages for the `cargo metadata` command.

See #12818 

### How should we test and review this PR?

Run `cargo metadata` to see that the `checksum` field is present. Tests have been updated with existing calculated checksums to reflect the change.

### Additional information

When running `cargo test` locally there are 2 tests failing, in particular

* [`alt_registry::alt_reg_metadata`](https://github.com/rust-lang/cargo/blob/master/tests/testsuite/alt_registry.rs#L810)
* [`alt_registry::unknown_registry`](https://github.com/rust-lang/cargo/blob/master/tests/testsuite/alt_registry.rs#L1122)

Running them on their own, e.g. `cargo test alt_registry::alt_reg_metadata` passes the test.

I will try to pinpoint the failing tests, maybe someone has an idea?

For example running test file `cargo test alt_registry` repeatedly will almost always fail the `unknown_registry` test. With a lower frequency it will additionally fail `alt_reg_metadata`.
